src/wp-admin/includes/image.php

```diff
@@ -1143,3 +1268,58 @@ function _copy_image_file( $attachment_id ) {
 
 	return $dst_file;
 }
+
+/**
+ * Returns an array with the list of valid mime types that a specific mime type should be converted into.
+ * For example an `image/jpeg` should be converted into an `image/jpeg` and `image/webp`. The first type
+ * is considered the primary output type for this image.
+ *
+ * @since 6.0.0
+ *
+ * @param $attachment_id int The attachment ID.
+ * @return array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function wp_upload_image_mime_transforms( $attachment_id ) {
+	$image_mime_transforms = array(
+		'image/jpeg'      => array( 'image/jpeg', 'image/webp' ),
+		'image/webp'      => array( 'image/webp', 'image/jpeg' ),
+		'application/pdf' => array( 'image/jpeg' ),
+	);
+
+	/**
+	 * Filter to the output mime types for a given input mime type.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $image_mime_transforms A map with the valid mime transforms.
+	 * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
+	 */
+	return (array) apply_filters( 'wp_upload_image_mime_transforms', $image_mime_transforms, $attachment_id );
+}
+
+/**
+ * Extract the primary and additional mime output types for an image from the $image_mime_transforms.
+ *
+ * @since 6.0.0
```

Comment:
Remember to add this to "private" functions:
```suggestion
 * @since 6.0.0
 * @access private
```